libs/datasets/dataset_orchestrator.py

```diff
@@ -0,0 +1,318 @@
+    "iso1:us#iso2:us-vi",
+    "iso1:us#iso2:us-as",
+    "iso1:us#iso2:us-gu",
+    # Subregion of AS
+    "iso1:us#iso2:us-vi#fips:78030",
+    "iso1:us#iso2:us-vi#fips:78020",
+    "iso1:us#iso2:us-vi#fips:78010",
+    # Retired FIPS
+    "iso1:us#iso2:us-sd#fips:46113",
+    "iso1:us#iso2:us-va#fips:51515",
+    # All the unknown county FIPS
+    *[pipeline.fips_to_location_id(f) for f in ABBREV_US_UNKNOWN_COUNTY_FIPS.values()],
+]
+
+
+@dataclass
+class OneStateDataset:
+    """Wrapper class containing a MultiRegionDataset with data for a single state and its subregions. 
+    
+    As consistent with the Region class, territories are considered states (e.g. GU, PR)
+    """
+
+    multiregion_dataset: MultiRegionDataset
+    state: pipeline.Region
+
+    @classmethod
+    def from_mrds(cls, dataset: MultiRegionDataset) -> "OneStateDataset":
+        states = [pipeline.Region.from_location_id(loc).state for loc in dataset.location_ids]
+        if len(set(states)) != 1:
+            raise ValueError(
+                "Can only create a OneStateDataset from datasets with data "
+                f"for a single state. Data for {len(set(states))} found."
+            )
+        return OneStateDataset(
+            multiregion_dataset=dataset, state=pipeline.Region.from_state(states[0])
+        )
+
+
+@dataclass
+class MultiRegionOrchestrator:
+    """Class to orchestrate dataset updates through state-focused parallelization."""
+
+    regions: Iterable[OneStateDataset]
+    cbsa_aggregator: statistical_areas.CountyToCBSAAggregator
+    region_overrides: Dict
+    print_stats: bool
+
+    @classmethod
+    def from_bulk_mrds(
+        cls,
+        states: Optional[List[str]] = None,
+        refresh_datasets: Optional[bool] = True,
+        print_stats: bool = True,
+    ) -> "MultiRegionOrchestrator":
+        bulk_dataset = load_bulk_dataset(refresh_datasets=refresh_datasets)
+        if not states:
+            states = list(US_STATE_ABBREV.values())
+        regions = [
+            OneStateDataset.from_mrds(bulk_dataset.get_subset(state=region)) for region in states
+        ]
+
+        cbsa_aggregator = statistical_areas.CountyToCBSAAggregator.from_local_public_data()
+        region_overrides = json.load(open(REGION_OVERRIDES_JSON))
+        return MultiRegionOrchestrator(
+            regions=regions,
+            region_overrides=region_overrides,
+            cbsa_aggregator=cbsa_aggregator,
+            print_stats=print_stats,
+        )
+
+    def build_and_combine_regions(
+        self, aggregate_to_country: bool = True, generate_cbsas: bool = True
+    ):
+        processed_regions = parallel_map(self._build_region_timeseries, self.regions)
+        _log.info("Finished processing individual regions...")
+        return self._combine_regions(
+            list(processed_regions),
+            aggregate_to_country=aggregate_to_country,
+            generate_cbsas=generate_cbsas,
+        )
+
+    def _combine_regions(
```

Comment:
nit: I might move _build_region_timeseries up above _combine_regions since it's used first in the above code?  Makes it a bit easier to read the code top-to-bottom.